### ISSUE TITLE
CF push fail with Error uploading files

### ISSUE THREAD

**Issue by rmahajan2420** [cloudfoundry/cli#813]
### Command
cf push dora

### CLI Version

cf version 6.16.1+924508c-2016-02-26

### Error
Error uploading application.
An unknown error occurred.
FAILED

### api vm has below error:
/var/vcap/sys/log/cloud_controller_worker_ctl.log:
### [2016-04-08 03:00:00+0000] [Worker(cc_api_worker.api_z1.0.2)] Job VCAP::CloudController::Jobs::ExceptionCatchingJob (id=257) FAILED (0 prior attempts) with CloudController::Blobstore::BlobstoreError: Could not create object, 404/<html>
[2016-04-08 03:00:00+0000] <head><title>404 Not Found</title></head>
[2016-04-08 03:00:00+0000] <body bgcolor=white>
[2016-04-08 03:00:00+0000] <center><h1>404 Not Found</h1></center>
[2016-04-08 03:00:00+0000] <hr><center>nginx</center>
[2016-04-08 03:00:00+0000] </body>
[2016-04-08 03:00:00+0000] </html>
[2016-04-08 03:00:00+0000] [Worker(cc_api_worker.api_z1.0.2)] Job VCAP::CloudController::Jobs::ExceptionCatchingJob (id=257) REMOVED permanently because of 1 consecutive failures
[2016-04-08 03:00:00+0000] [Worker(cc_api_worker.api_z1.0.2)] 1 jobs processed at 4.4393 j/s, 1 failed

### CC API Version
API version: 2.52.0

### CF Trace
Attached 
[push-error.log.zip](https://github.com/cloudfoundry/cli/files/209586/push-error.log.zip)

### Platform Details

e.g. amazon linux ami

[ec2-user@ip- micro-deployment]$ cf events dora
Getting events for app dora in org default_organization / space default_space as admin...

time                          event                 actor   description   
2016-04-08T03:03:09.00+0000   audit.app.update      admin      
2016-04-08T02:59:58.00+0000   audit.app.update      admin      
2016-04-08T02:19:19.00+0000   audit.app.update      admin      
2016-04-08T02:16:42.00+0000   audit.app.update      admin      
2016-04-08T02:14:54.00+0000   audit.app.update      admin      
2016-04-08T02:14:47.00+0000   audit.app.update      admin      
2016-04-08T02:14:20.00+0000   audit.app.update      admin      
2016-04-08T02:12:04.00+0000   audit.app.update      admin      
2016-04-08T02:12:04.00+0000   audit.app.map-route   admin      
2016-04-08T02:12:04.00+0000   audit.app.create      admin   instances: 1, state: STOPPED, environment_json: PRIVATE DATA HIDDEN   

Please help

**Comment by dkoper** [comment]
@SocalNick, @dieucao: CC is returning an unknown error. Any ideas?

**Comment by rmahajan2420** [comment]
Just to add one more thing here.. blobstore_z1/0 VM's access log is complaining about 404's:

 - blobstore-username [08/Apr/2016:04:59:28 +0000] "PUT /admin/cc-resources/d3/9d/d39d0c7557642754a4641ff3a14aa8a7a9253260 HTTP/1.1" 404 162 "-" "HTTPClient/1.0 (2.7.1, ruby 2.2.4 (2015-12-16))" 0.000 - .

**Comment by SocalNick** [comment]
@rmahajan2420 could you post your blobstore configuration. please be sure to remove any sensitive information.

**Comment by rmahajan2420** [comment]
Hi @SocalNick thanks for your reply. Please see below:
- name: blobstore_z1
  instances: 1
  persistent_disk: 102400
  resource_pool: small_z1
  templates:
  - {name: blobstore, release: cf}
  - {name: metron_agent, release: cf}
  - {name: route_registrar, release: cf}
  - {name: consul_agent, release: cf}
  networks:
  - name: cf_private
  properties:
    consul:
      agent:
        services:
          blobstore: {}
    route_registrar:
      routes:
      - name: blobstore
        port: 80
        registration_interval: 20s
        tags:
          component: blobstore
        uris:
        - "blobstore.<ELASTIC IP of ha-proxy>.xip.io"
-------
blobstore:
    admin_users:
    - password: <-->
      username: <--->
    secure_link:
      secret: <--->
    tls:
      cert: <ca.cert>
      private_key: <ca.key>
        
Is that what you are asking? FYI, i am referencing this link and using minimal aws install:  https://github.com/cloudfoundry/cf-release/tree/master/example_manifests

Awaiting for your reply. Thanks heaps for looking into this. Let me know if you need further information.
Ricky

**Comment by SocalNick** [comment]
@dkoper I moved this to CF CAPI Public tracker.
@rmahajan2420 I'll have someone on my team take a look

**Comment by adowns01** [comment]
Hi @rmahajan2420, 

It looks like you are using an old bosh-lite manifest. We don't use the `.xip.io` for the system addresses anymore. That's the only difference I see here, but there are probably more throughout the rest of your manifest. 

I recommend
- download [latest stemcell](http://bosh.io/)
- update cf-release
- update bosh-lite
- deploy bosh lite following these direction with [their new provisioning script](https://github.com/cloudfoundry/bosh-lite#deploy-cloud-foundry)

Please let us know if you are still having issues.
@adowns01 & @simonleung8, CAPI Team Members

**Comment by rmahajan2420** [comment]
Hi @adowns01.. Thanks for looking into this : As suggested I have done the update. I can see all of the vms are running now. However cf push dora is giving me  "FAILED
Unauthorized error: You are not authorized. Error: Invalid authorization" exception.

I can see still see the route to that app is going through api.<ELP of Ha proxy>.xip.io..
cf apps
Getting apps in org default_organization / space default_space as admin...
OK
name   requested state   instances   memory   disk   urls   
dora   started           0/1         1G       1G     dora.<SYSTEMDOMAIN>.xip.io.

Please help..

Thanks,
Ricky

**Comment by SocalNick** [comment]
@rmahajan2420 apologies for the delay on this. Have you made any progress?

I haven't installed bosh-lite on AWS in sometime, so I did it today. I followed [the instructions in bosh-lite repo](https://github.com/cloudfoundry/bosh-lite/blob/master/docs/aws-provider.md#using-the-aws-provider) to get my bosh-lite running.

I uploaded [CF-235](http://bosh.io/releases/github.com/cloudfoundry/cf-release?version=235) and [the latest warden stemcell](http://bosh.io/stemcells/bosh-warden-boshlite-ubuntu-trusty-go_agent).

Then I checked out the CF-235 tag in cf-release and generated a manifest using `./scripts/generate-bosh-lite-dev-manifest stub-for-bosh-lite`. The contents of `stub-for-bosh-lite` were:

```
---
director_uuid: 04f2c9f0-****-****-****-2e34f88b3854
properties:
  domain: 54.***.***.***.xip.io
```

Then I just did `bosh deploy` and everything worked.
